js/DisableUiComponent.js

```diff
@@ -0,0 +1,56 @@
+'use strict';
+
+var $ = require('jquery');
+
+function DisableUiComponent(html) {
+    this.$html = html;
+}
+
+function preventDefaultAndStopPropagation(e) {
+    e.preventDefault();
+    e.stopPropagation();
+}
+
+DisableUiComponent.prototype.init = function () {
+    var component = this;
+
+    component.$container = this.$html.find('[data-disable-ui="true"]');
+
+    component.$container.each(function() {
+
+        var $this = $(this),
+            FORM_ELEMENTS = 'button:not(.disabled), input:not(.disabled), select:not(.disabled)',
+            LINK_ELEMENTS = 'a:not(.disabled)',
+            LABEL_ELEMENTS = 'label';
+
+        // Disable form elements
+        $this.find(FORM_ELEMENTS)
+            .on('click', preventDefaultAndStopPropagation)
+            .addClass('disabled')
+            .attr('disabled', 'disabled');
+
+        // Disable labels
+        $this.find(LABEL_ELEMENTS).addClass('u-cursor-not-allowed');
```

Comment:
is this not something the `.is-disabled` class could handle? We use that quite often and it's a common state variation for a given element, or a group of elements. Same for all instances of `.u-cursor-not-allowed`.